internal/operators/ocs/validations.go

```diff
@@ -0,0 +1,209 @@
+				if err != nil {
+					o.log.Fatal("Error occured while calculating Node requirements ", err)
+					return false
+				}
+			}
+		}
+	}
+	// Hosts with disks must be a multiple of 3
+	o.OCSDisksAvailable = HostsWithDisks / 3
+	canDeployOCS, status := o.validateOCS(o.log, hosts, CPUCount, TotalRAM, DiskCount, HostsWithDisks)
+
+	err := SetOperatorStatus(cluster, status)
+	if err != nil {
+		o.log.Fatal("Error occured while Marshalling or Unmarshaling Operators ", err)
+		return false
+	}
+
+	return canDeployOCS
+}
+
+func (o *ocsValidator) nodeRequirements(host *models.Host, CPUCount *int64, TotalRAM *int64, DiskCount *int64, HostsWithDisks *int64) error {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
+		return err
+	}
+
+	disks, err := o.hostApi.GetHostValidDisks(host)
+	if err != nil {
+		o.log.Error("failed to get valid disks due to ", err.Error())
+		return err
+	}
+
+	if len(disks) > 1 { // OCS must use the non-boot disks
+		*DiskCount += (int64)(len(disks) - 1) // not counting the boot disk
+		*HostsWithDisks++
+	}
+
+	*CPUCount += inventory.CPU.Count
+	*TotalRAM += inventory.Memory.UsableBytes
+	return nil
+}
+
+func gbToBytes(gb int64) int64 {
+	return gb * int64(units.GB)
+}
+
+func (o *ocsValidator) validateOCS(log logrus.FieldLogger, hosts []*models.Host, cpu int64, RAM int64, disk int64, hostsWithDisks int64) (bool, string) {
+
+	var status string
+	if int64(len(hosts)) == o.OCSRequiredHosts { // for 3 hosts
+		if cpu < o.OCSRequiredCompactModeCPUCount*3 || RAM < gbToBytes(o.OCSRequiredCompactModeRAMGB) || disk < o.OCSRequiredDisk || hostsWithDisks < o.OCSRequiredHosts { // check for master nodes requirements
+			status = "The resources are not sufficient to deploy OCS on three hosts. "
+			if cpu < o.OCSRequiredCompactModeCPUCount*3 {
+				status = status + fmt.Sprint("A minimum of ", o.OCSRequiredCompactModeCPUCount*3, " CPUs, ")
+			}
+			if RAM < gbToBytes(o.OCSRequiredCompactModeRAMGB) {
+				status = status + fmt.Sprint("A minimum of ", o.OCSRequiredCompactModeRAMGB, " RAM, ")
+			}
+			if disk < o.OCSRequiredDisk {
+				status = status + fmt.Sprint("A minimum of ", o.OCSRequiredDisk, " Disks, ")
+			}
+			if hostsWithDisks < o.OCSRequiredHosts {
+				status = status + fmt.Sprint("A minimum of ", o.OCSRequiredHosts, " Hosts with disks, ")
+			}
+			status = status + "are required."
+			log.Info(status)
+			return false, status
+		}
+		status = "OCS Requirements for Compact Mode are satisfied"
+		return true, status
+	}
+	if disk < o.OCSRequiredDisk || cpu < o.OCSMinimumCPUCount || RAM < gbToBytes(o.OCSMinimumRAMGB) || hostsWithDisks < o.OCSRequiredHosts { // check for worker nodes requirements
+		status = "The resources are not sufficient to deploy OCS on worker hosts. "
+		if cpu < o.OCSMinimumCPUCount {
+			status = status + fmt.Sprint("A minimum of ", o.OCSMinimumCPUCount, " CPUs, ")
```

Comment:
"A minimum of" can be removed from these strings.

Reply:
fixed